subprojects/tooling-api-builders/src/main/java/org/gradle/tooling/internal/provider/runner/TestExecutionResultEvaluator.java

```diff
@@ -100,6 +102,22 @@ private String formatInternalTestExecutionRequest() {
                 requestDetails.append("\n").append(Strings.repeat(INDENT, 2)).append("Test method ").append(className).append(".").append(methodName).append("()");
             }
         }
+
+        for (InternalTestPatternSpec testPatternSpec : internalTestExecutionRequest.getTestPatternSpecs()) {
+            for (String cls : testPatternSpec.getClasses()) {
+                requestDetails.append("\n").append(Strings.repeat(INDENT, 2)).append("Test class: ").append(cls).append(" in task " + testPatternSpec.getTaskPath());
+            }
+            for (Map.Entry<String, List<String>> method : testPatternSpec.getMethods().entrySet()) {
+                requestDetails.append("\n").append(Strings.repeat(INDENT, 2)).append("Test method ").append(method.getKey()).append(".").append(method.getValue()).append("()").append(" in task " + testPatternSpec.getTaskPath());
+            }
+            for (String pkg : testPatternSpec.getPackages()) {
+                requestDetails.append("\n").append(Strings.repeat(INDENT, 2)).append("Test package ").append(pkg).append(" in task " + testPatternSpec.getTaskPath());
+            }
+            for (String pattern : testPatternSpec.getPatterns()) {
+                requestDetails.append("\n").append(Strings.repeat(INDENT, 2)).append("Test pattern ").append(pattern).append(" in task " + testPatternSpec.getTaskPath());
+            }
+        }
+
```

Comment:
Do you have some example of where this shows up in the output?

Reply:
I've added an separate test case to `TestLauncherTestSpecCrossVersionSpec` to verify the output.